ModelReplacementAPI/Modules/AvatarUpdater.cs

```diff
@@ -67,61 +65,63 @@ protected virtual void UpdateModel()
                 RotationOffset offset = modelBone.GetComponent<RotationOffset>();
                 if (offset) { modelBone.rotation *= offset.offset; }
             }
+
             Transform rootBone = GetAvatarTransformFromBoneName("spine");
             Transform playerRootBone = GetPlayerTransformFromBoneName("spine");
             rootBone.position = playerRootBone.position + playerRootBone.TransformVector(rootPositionOffset);
         }
+
         public virtual void Update()
         {
-            if (playerModelRenderer == null) { return; }
-            if (replacementAnimator == null) { return; }
+            if (playerModelRenderer == null || replacementAnimator == null) { return; }
             UpdateModel();
         }
 
         public Transform GetAvatarTransformFromBoneName(string boneName)
         {
-            //Special logic is required here. The player model has 5 central bones.
-            // Spine, spine.001, spine.002, spine.003,   spine.004, corresponding to 
-            // Hips   Spine      Chest      UpperChest   Head
-            //However spine.002 practically doesn't move, and I wish to support mods that don't have an UpperChest bone. 
-            //If they don't have an upperchest bone, instead map spine.003 to the Chest transform on the replacement model.
             if (boneName == "spine.002")
             {
-                if (hasUpperChest) { return replacementAnimator.GetBoneTransform(HumanBodyBones.Chest); }
-                else { return null; }
+                return hasUpperChest ? replacementAnimator.GetBoneTransform(HumanBodyBones.Chest) : null;
             }
+
             if (boneName == "spine.003")
             {
-                if (hasUpperChest) { return replacementAnimator.GetBoneTransform(HumanBodyBones.UpperChest); }
-                else { return replacementAnimator.GetBoneTransform(HumanBodyBones.Chest); }
+                return hasUpperChest ? replacementAnimator.GetBoneTransform(HumanBodyBones.UpperChest) : replacementAnimator.GetBoneTransform(HumanBodyBones.Chest);
             }
-            if (boneName.Contains("PlayerRagdoll")) // Ragdoll Hips
+
+            if (boneName.Contains("PlayerRagdoll"))
             {
                 return replacementAnimator.GetBoneTransform(HumanBodyBones.Hips);
             }
-            if (modelToAvatarBone.ContainsKey(boneName))
-            {
-                return replacementAnimator.GetBoneTransform(modelToAvatarBone[boneName]);
-            }
-            return null;
+
+            return modelToAvatarBone.TryGetValue(boneName, out HumanBodyBones avatarBone)
+                ? replacementAnimator.GetBoneTransform(avatarBone)
+                : null; // throw new Exception($"Failed to find bone {boneName}");
```

Comment:
There seems to be unlisted hips bones passed in this argument, but since you ignore them I guess I can do it too

Reply:
This relates to the skeletal structure of the player ragdoll, which instead of "Spine" uses "PlayerRagdoll" as the hip bone.